llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -156,14 +156,21 @@ static bool resultDependsOnExec(const MachineInstr &MI) {
     Register DstReg = MI.getOperand(0).getReg();
     if (!DstReg.isVirtual())
       return true;
-    for (MachineInstr &Use : MRI.use_nodbg_instructions(DstReg)) {
-      switch (Use.getOpcode()) {
+    for (MachineOperand &Use : MRI.use_nodbg_operands(DstReg)) {
+      MachineInstr &User = *Use.getParent();
+      switch (User.getOpcode()) {
       case AMDGPU::S_AND_SAVEEXEC_B32:
       case AMDGPU::S_AND_SAVEEXEC_B64:
         break;
+      case AMDGPU::SI_IF:
+      case AMDGPU::SI_IF_BREAK:
+        if (Use.getOperandNo() != 1)
```

Comment:
Presumably this branch is specific to SI_IF_BREAK?
Could you add extra comment to explain what operand 1 is.